In [2]:
import requests
from bs4 import BeautifulSoup

In [3]:
import pandas as pd
import json
import numpy as np

In [4]:
url_endpoint = "https://www.allkeyshop.com/blog/wp-admin/admin-ajax.php"
reqParams = {
    'action': 'catalogue_query',
    'query[search]': 'hitman 3',
    'query[productsPerPage]': 16,
    'query[currency]': 'eur',
    'query[category]': 'pc-games-all'
}

In [5]:
resp = requests.get(url_endpoint, params=reqParams).json()['results']
soup = BeautifulSoup(resp, 'html.parser')
productId = soup.find(class_="metacritic-button-green")['data-product-id']
productId

'38888'

In [6]:
reqParams = {
    'action': 'get_offers',
    'product': productId,
    'currency': 'eur',
    'use_beta_offers_display': 1
}

payload = {}

response = requests.request(
    "GET", url_endpoint, params=reqParams, data=payload)
response.json()

{'success': True,
 'offers': [{'id': 134504270,
   'affiliateUrl': 'https://www.hrkgame.com/en/games/product/hitman-3-europe',
   'isActive': True,
   'merchant': '94',
   'price': {'eur': {'bestCoupon': {'code': '-10%: ALLKEYSHOP10',
      'discountValue': '10.00',
      'discountStrategy': '%',
      'isCashback': False},
     'currency': 'eur',
     'price': 15.48,
     'priceWithoutCoupon': 17.2}},
   'edition': '1',
   'region': '80eu',
   'stock': 'InStock',
   'platform': 'epic-store'},
  {'id': 134474250,
   'affiliateUrl': 'https://www.eneba.com/epic-games-hitman-3-epic-games-key-europe?af_id=allkeyshop',
   'isActive': True,
   'merchant': '272',
   'price': {'eur': {'bestCoupon': {'code': '8%: AKS8',
      'discountValue': '8.00',
      'discountStrategy': '%',
      'isCashback': False},
     'currency': 'eur',
     'price': 15.63,
     'priceWithoutCoupon': 16.99}},
   'edition': '1',
   'region': '80eu',
   'stock': 'InStock',
   'platform': 'epic-store'},
  {'id': 133395

In [7]:
#f=open('test.json','r')
#file=json.loads(f.read())
file=response.json()

In [8]:
file.keys()

dict_keys(['success', 'offers', 'merchants', 'editions', 'regions'])

In [9]:
file['offers'][0]['price']['eur'].keys()

dict_keys(['bestCoupon', 'currency', 'price', 'priceWithoutCoupon'])

In [10]:
file['offers'][0]['price']['eur']['priceWithoutCoupon']

17.2

In [11]:
data=pd.json_normalize(file,record_path='offers')
data

id                                       affiliateUrl  isActive  \
0   134504270  https://www.hrkgame.com/en/games/product/hitma...      True   
1   134474250  https://www.eneba.com/epic-games-hitman-3-epic...      True   
2   133395388  https://www.kinguin.net/category/80607/hitman-...      True   
3   134522273  https://k4g.com/product/hitman-3-epic-games-eu...      True   
4   133299847  https://www.gamivo.com/product/hitman-3-eu-epi...      True   
..        ...                                                ...       ...   
86  134638686  https://www.hrkgame.com/en/games/product/hitma...      True   
87  134634311  https://www.cdkeys.com/hitman-3-deluxe-edition...      True   
88  134530249  https://keyesd.myshopify.com/products/hitman-3...      True   
89  133534742             https://punktid.com/hitman-3-greencode      True   
90  133398774  https://www.gamingdragons.com/en/game/buy-hitm...      True   

   merchant edition    region       stock    platform  \
0        94       1      80eu     InStock  epic-store   
1       272       1      80eu     InStock  epic-store   
2        47       1      80eu     InStock  epic-store   
3       324       1      80eu     InStock  epic-store   
4       218       1      80eu     InStock  epic-store   
..      ...     ...       ...         ...         ...   
86       94       7  steamrow  OutOfStock       steam   
87        9       7         2  OutOfStock       steam   
88      325       1        80  OutOfStock  epic-store   
89      307       1       634  OutOfStock  epic-store   
90       73       7        80  OutOfStock  epic-store   

   price.eur.bestCoupon.code price.eur.bestCoupon.discountValue  \
0         -10%: ALLKEYSHOP10                              10.00   
1                   8%: AKS8                               8.00   
2             14%: OCTOBER14                              14.00   
3          10%: AKS10OCTOBER                              10.00   
4             8%: AKSOCTOBER                               8.00   
..                       ...                                ...   
86        -10%: ALLKEYSHOP10                              10.00   
87                       NaN                                NaN   
88                       NaN                                NaN   
89                 -5%: AKS5                               5.00   
90                       NaN                                NaN   

   price.eur.bestCoupon.discountStrategy price.eur.bestCoupon.isCashback  \
0                                      %                           False   
1                                      %                           False   
2                                      %                           False   
3                                      %                           False   
4                                      %                           False   
..                                   ...                             ...   
86                                     %                           False   
87                                   NaN                             NaN   
88                                   NaN                             NaN   
89                                     %                           False   
90                                   NaN                             NaN   

   price.eur.currency  price.eur.price  price.eur.priceWithoutCoupon  \
0                 eur            15.48                         17.20   
1                 eur            15.63                         16.99   
2                 eur            15.65                         18.20   
3                 eur            15.65                         17.39   
4                 eur            16.02                         17.41   
..                ...              ...                           ...   
86                eur            30.05                         33.39   
87                eur            31.19                         31.19   
88                eur  

In [12]:
mapp=data.drop(['price.eur.bestCoupon.discountValue','price.eur.bestCoupon.discountStrategy','price.eur.bestCoupon.isCashback','price.eur.bestCoupon.code', 'price.eur.bestCoupon'], axis=1,inplace=True)

In [13]:
data

id                                       affiliateUrl  isActive  \
0   134504270  https://www.hrkgame.com/en/games/product/hitma...      True   
1   134474250  https://www.eneba.com/epic-games-hitman-3-epic...      True   
2   133395388  https://www.kinguin.net/category/80607/hitman-...      True   
3   134522273  https://k4g.com/product/hitman-3-epic-games-eu...      True   
4   133299847  https://www.gamivo.com/product/hitman-3-eu-epi...      True   
..        ...                                                ...       ...   
86  134638686  https://www.hrkgame.com/en/games/product/hitma...      True   
87  134634311  https://www.cdkeys.com/hitman-3-deluxe-edition...      True   
88  134530249  https://keyesd.myshopify.com/products/hitman-3...      True   
89  133534742             https://punktid.com/hitman-3-greencode      True   
90  133398774  https://www.gamingdragons.com/en/game/buy-hitm...      True   

   merchant edition    region       stock    platform price.eur.currency  \
0        94       1      80eu     InStock  epic-store                eur   
1       272       1      80eu     InStock  epic-store                eur   
2        47       1      80eu     InStock  epic-store                eur   
3       324       1      80eu     InStock  epic-store                eur   
4       218       1      80eu     InStock  epic-store                eur   
..      ...     ...       ...         ...         ...                ...   
86       94       7  steamrow  OutOfStock       steam                eur   
87        9       7         2  OutOfStock       steam                eur   
88      325       1        80  OutOfStock  epic-store                eur   
89      307       1       634  OutOfStock  epic-store                eur   
90       73       7        80  OutOfStock  epic-store                eur   

    price.eur.price  price.eur.priceWithoutCoupon  
0             15.48                         17.20  
1             15.63                         16.99  
2             15.65                         18.20  
3             15.65                         17.39  
4             16.02                         17.41  
..              ...                           ...  
86            30.05                         33.39  
87            31.19                         31.19  
88            33.04                         33.04  
89            47.02                         49.49  
90            67.95                         67.95  

[91 rows x 11 columns]

In [14]:
data.rename(columns = {'price.eur.currency':'currency', 'price.eur.price':'price','price.eur.priceWithoutCoupon':'priceWithoutCoupon'}, inplace = True)

In [15]:
data[data.priceWithoutCoupon==data.priceWithoutCoupon.min()]

id                                       affiliateUrl  isActive  \
5  132997844  https://www.instant-gaming.com/en/6851-buy-key...      True   

  merchant edition region    stock    platform currency  price  \
5       13       1     80  InStock  epic-store      eur  16.69   

   priceWithoutCoupon  
5               16.69

In [16]:
merchants_list=pd.DataFrame(data=file['merchants']).T
merchants_list

id            name                 aggregateRating           types  \
94        94             HRK   {'value': 2.71, 'count': 321}    CD Key Store   
272      272           Eneba   {'value': 3.62, 'count': 313}    CD Key Store   
47        47         Kinguin   {'value': 4.17, 'count': 586}    CD Key Store   
324      324             K4G   {'value': 4.85, 'count': 664}    CD Key Store   
218      218          GAMIVO  {'value': 4.17, 'count': 1237}    CD Key Store   
13        13  Instant Gaming   {'value': 4.26, 'count': 306}    CD Key Store   
9          9      CDKeys.com   {'value': 4.17, 'count': 724}    CD Key Store   
77        77   Gamers Outlet   {'value': 4.09, 'count': 151}    CD Key Store   
224      224      Vidaplayer    {'value': 3.59, 'count': 11}    CD Key Store   
61616  61616        G2A Plus   {'value': 4.29, 'count': 292}    CD Key Store   
40        40           Mmoga    {'value': 3.2, 'count': 180}    CD Key Store   
67        67      CJS-CDKeys   {'value': 3.79, 'count': 803}    CD Key Store   
73        73   Gamingdragons    {'value': 3.04, 'count': 69}    CD Key Store   
61        61             G2A   {'value': 4.29, 'count': 292}    CD Key Store   
49        49       Play-Asia    {'value': 4.55, 'count': 10}    CD Key Store   
307      307         Punktid    {'value': 2.21, 'count': 33}    CD Key Store   
315      315    Livecards EU     {'value': 3.58, 'count': 6}    CD Key Store   
325      325          Keyesd   {'value': 4.73, 'count': 123}    CD Key Store   
321      321   Royal CD Keys    {'value': 4.4, 'count': 286}    CD Key Store   
222      222        Voidu EU   {'value': 3.89, 'count': 100}  Official Store   
157      157      Epic Games     {'value': 3.06, 'count': 9}  Official Store   
63        63  Greenmangaming    {'value': 4.61, 'count': 46}  Official Store   
126      126    Humble Store        {'value': 5, 'count': 5}  Official Store   
1          1           Steam     {'value': 4.8, 'count': 46}  Official Store   

                                         paymentMethods        logoSlug  \
94     {'paymentMethode': ['card', 'paypal', 'crypto']}             hrk   
272              {'paymentMethode': ['card', 'paypal']}           eneba   
47               {'paymentMethode': ['card', 'paypal']}         kinguin   
324              {'paymentMethode': ['card', 'paypal']}             k4g   
218    {'paymentMethode': ['card', 'paypal', 'crypto']}          gamivo   
13               {'paymentMethode': ['paypal', 'card']}  instant-gaming   
9                {'paymentMethode': ['card', 'paypal']}      cdkeys-com   
77               {'paymentMethode': ['card', 'paypal']}   gamers-outlet   
224    {'paymentMethode': ['card', 'crypto', 'paypal']}      vidaplayer   
61616            {'paymentMethode': ['card', 'paypal']}             g2a   
40     {'paymentMethode': ['card', 'paypal', 'crypto']}           mmoga   
67               {'paymentMethode': ['card', 'paypal']}      cjs-cdkeys   
73               {'paymentMethode': ['card', 'paypal']}   gamingdragons   
61               {'paymentMethode': ['card', 'paypal']}             g2a   
49               {'paymentMethode': ['card', 'paypal']}       play-asia   
307            {'paymentMethode': ['paypal', 'crypto']}         punktid   
315              {'paymentMethode': ['card', 'crypto']}       livecards   
325                        {'paymentMethode': ['card']}          keyesd   
321                        {'paymentMethode': ['card']}   royal-cd-keys   
222              {'paymentMethode': ['card', 'paypal']}           voidu   
157              {'paymentMethode': ['card', 'paypal']}      epic-games   
63     {'paymentMethode': ['card', 'paypal', 'crypto']}  greenmangaming   
126              {'paymentMethode': ['card', 'paypal']}    humble-store   
1                {'paymentMethode': ['card', 'paypal']}           steam   

                                               reviewUrl  
94       https://www.allkeyshop.com/blog/review/hrkgame/  
272        http

In [140]:
merged_list = pd.merge(data,merchants_list[['id','name']],left_on='merchant',right_on='id',how='left')
merged_list.drop(['id_y','price'], axis=1,inplace=True)
merged_list.rename(columns = {'priceWithoutCoupon':'price'}, inplace = True)
#merged_list.set_index('id_x',inplace=True)

In [141]:
merged_list.head(-12)

id_x                                       affiliateUrl  isActive  \
0   134504270  https://www.hrkgame.com/en/games/product/hitma...      True   
1   134474250  https://www.eneba.com/epic-games-hitman-3-epic...      True   
2   133395388  https://www.kinguin.net/category/80607/hitman-...      True   
3   134522273  https://k4g.com/product/hitman-3-epic-games-eu...      True   
4   133299847  https://www.gamivo.com/product/hitman-3-eu-epi...      True   
..        ...                                                ...       ...   
74  134564612  https://greenmangaming.sjv.io/c/1297091/127200...      True   
75  134554781  https://greenmangaming.sjv.io/c/1297091/127200...      True   
76  134668913  https://www.epicgames.com/store/p/hitman-3--de...      True   
77  134658895  https://www.voidu.com/en/product/hitman-3-delu...      True   
78  133397371  https://www.humblebundle.com/store/hitman-3-de...      True   

   merchant edition    region    stock    platform currency  price  \
0        94       1      80eu  InStock  epic-store      eur  17.20   
1       272       1      80eu  InStock  epic-store      eur  16.99   
2        47       1      80eu  InStock  epic-store      eur  18.20   
3       324       1      80eu  InStock  epic-store      eur  17.39   
4       218       1      80eu  InStock  epic-store      eur  17.41   
..      ...     ...       ...      ...         ...      ...    ...   
74       63       7  steamrow  InStock       steam      eur  89.99   
75       63       7        80  InStock  epic-store      eur  89.99   
76      157       7        80  InStock  epic-store      eur  89.99   
77      222       7         2  InStock       steam      eur  89.99   
78      126       7        80  InStock  epic-store      eur  89.99   

              name  
0              HRK  
1            Eneba  
2          Kinguin  
3              K4G  
4           GAMIVO  
..             ...  
74  Greenmangaming  
75  Greenmangaming  
76      Epic Games  
77        Voidu EU  
78    Humble Store  

[79 rows x 11 columns]

In [142]:
instock_list=merged_list.loc[(merged_list['platform'] == 'steam') & (merged_list['stock'] == 'InStock')]

In [143]:
instock_list['price'].describe()

count     48.000000
mean      52.379792
std       24.557060
min       21.950000
25%       30.760000
50%       49.440000
75%       63.150000
max      114.030000
Name: price, dtype: float64

In [144]:
table=instock_list[["name", "price"]].groupby("name").min()
table.sort_values('price', ascending=True,inplace=True)

In [145]:
table.head(5)

price
name                 
Eneba           21.95
Instant Gaming  25.75
CDKeys.com      25.79
Mmoga           26.99
Kinguin         27.90

In [146]:
instock_list['price'].min()

21.95

In [147]:
instock_list[instock_list.price==instock_list.price.min()]

id_x                                       affiliateUrl  isActive  \
15  134513555  https://www.eneba.com/steam-hitman-3-pc-steam-...      True   

   merchant edition region    stock platform currency  price   name  
15      272       1      2  InStock    steam      eur  21.95  Eneba

In [148]:
merged_list[merged_list.price==merged_list.price.min()]

id_x                                       affiliateUrl  isActive  \
5  132997844  https://www.instant-gaming.com/en/6851-buy-key...      True   

  merchant edition region    stock    platform currency  price            name  
5       13       1     80  InStock  epic-store      eur  16.69  Instant Gaming

In [26]:
# Building blocks for the URL
entrypoint = 'https://sdw-wsrest.ecb.europa.eu/service/' # Using protocol 'https'
resource = 'data'           # The resource for data queries is always'data'
flowRef ='EXR'              # Dataflow describing the data that needs to be returned, exchange rates in this case
key = 'D.SEK.EUR.SP00.A'    # Defining the dimension values, explained below

# Define the parameters
parameters = {
    'startPeriod': '2000-01-01',  # Start date of the time series
    'endPeriod': '2022-10-10'     # End of the time series
}

In [27]:
# Construct the URL: https://sdw-wsrest.ecb.europa.eu/service/data/EXR/D.CHF.EUR.SP00.A
request_url = entrypoint + resource + '/'+ flowRef + '/' + key

# Make the HTTP request again, now requesting for CSV format
response = requests.get(request_url, params=parameters, headers={'Accept': 'text/csv'})
# Response succesful? (Response code 200)
print(response)

<Response [200]>


In [28]:
open('temp.csv', 'wb').write(response.content)
currency_df = pd.read_csv('temp.csv')

In [29]:
#currency_df.info()

In [30]:
currency_df.tail(1)['OBS_VALUE']

5890    10.9502
Name: OBS_VALUE, dtype: float64

In [31]:
10.9502*16.69

182.75883800000003

In [139]:
ts = currency_df.filter(['TIME_PERIOD', 'OBS_VALUE'], axis=1)
# 'TIME_PERIOD' was of type 'object' (as seen in df.info). Convert it to datetime first
ts['TIME_PERIOD'] = pd.to_datetime(ts['TIME_PERIOD'])
# Set 'TIME_PERIOD' to be the index
ts = ts.set_index('TIME_PERIOD')
# Print the last 5 rows to screen
ts.tail(1)

OBS_VALUE
TIME_PERIOD           
2022-10-10     10.9502

In [138]:
rate_sek = currency_df.tail(1)['OBS_VALUE'].values[0]
rate_sek

10.9502

In [149]:
merged_list["priceSEK"] = merged_list["price"] * rate_sek

In [150]:
df2 = merged_list.assign(priceSEK=merged_list["price"]*rate_sek)
df2

id_x                                       affiliateUrl  isActive  \
0   134504270  https://www.hrkgame.com/en/games/product/hitma...      True   
1   134474250  https://www.eneba.com/epic-games-hitman-3-epic...      True   
2   133395388  https://www.kinguin.net/category/80607/hitman-...      True   
3   134522273  https://k4g.com/product/hitman-3-epic-games-eu...      True   
4   133299847  https://www.gamivo.com/product/hitman-3-eu-epi...      True   
..        ...                                                ...       ...   
86  134638686  https://www.hrkgame.com/en/games/product/hitma...      True   
87  134634311  https://www.cdkeys.com/hitman-3-deluxe-edition...      True   
88  134530249  https://keyesd.myshopify.com/products/hitman-3...      True   
89  133534742             https://punktid.com/hitman-3-greencode      True   
90  133398774  https://www.gamingdragons.com/en/game/buy-hitm...      True   

   merchant edition    region       stock    platform currency  price  \
0        94       1      80eu     InStock  epic-store      eur  17.20   
1       272       1      80eu     InStock  epic-store      eur  16.99   
2        47       1      80eu     InStock  epic-store      eur  18.20   
3       324       1      80eu     InStock  epic-store      eur  17.39   
4       218       1      80eu     InStock  epic-store      eur  17.41   
..      ...     ...       ...         ...         ...      ...    ...   
86       94       7  steamrow  OutOfStock       steam      eur  33.39   
87        9       7         2  OutOfStock       steam      eur  31.19   
88      325       1        80  OutOfStock  epic-store      eur  33.04   
89      307       1       634  OutOfStock  epic-store      eur  49.49   
90       73       7        80  OutOfStock  epic-store      eur  67.95   

             name    priceSEK  
0             HRK  188.343440  
1           Eneba  186.043898  
2         Kinguin  199.293640  
3             K4G  190.423978  
4          GAMIVO  190.642982  
..            ...         ...  
86            HRK  365.627178  
87     CDKeys.com  341.536738  
88         Keyesd  361.794608  
89        Punktid  541.925398  
90  Gamingdragons  744.066090  

[91 rows x 12 columns]

In [151]:
df2[df2.priceSEK==df2.priceSEK.min()]

id_x                                       affiliateUrl  isActive  \
5  132997844  https://www.instant-gaming.com/en/6851-buy-key...      True   

  merchant edition region    stock    platform currency  price  \
5       13       1     80  InStock  epic-store      eur  16.69   

             name    priceSEK  
5  Instant Gaming  182.758838

In [152]:
df2.loc[df2['priceSEK'].idxmin()]

id_x                                                    132997844
affiliateUrl    https://www.instant-gaming.com/en/6851-buy-key...
isActive                                                     True
merchant                                                       13
edition                                                         1
region                                                         80
stock                                                     InStock
platform                                               epic-store
currency                                                      eur
price                                                       16.69
name                                               Instant Gaming
priceSEK                                               182.758838
Name: 5, dtype: object

In [153]:
df2.nsmallest(2,'priceSEK').filter(['name','priceSEK'],axis=1)

name    priceSEK
5  Instant Gaming  182.758838
1           Eneba  186.043898

In [154]:
df2.loc[df2['platform']=='steam',['priceSEK','name']].sort_values(by='priceSEK').head(3)

priceSEK           name
15  240.356890          Eneba
85  250.102568  Gamers Outlet
84  256.891692            HRK

In [155]:
instock_list[instock_list.price==instock_list.price.min()]

id_x                                       affiliateUrl  isActive  \
15  134513555  https://www.eneba.com/steam-hitman-3-pc-steam-...      True   

   merchant edition region    stock platform currency  price   name  
15      272       1      2  InStock    steam      eur  21.95  Eneba

In [41]:
steamAppsResponse = requests.get(
    "https://api.steampowered.com/ISteamApps/GetAppList/v2/")

steamApps = steamAppsResponse.json()
steamApps = steamApps['applist']['apps']  # ['app']
#steamApps

In [157]:
steam_df=pd.json_normalize(steamApps)
#steam_df.set_index('appid',inplace=True)

steam_df.tail()
dfs1=steam_df

remove all empty names rows

In [212]:
dfs1.replace('', np.nan, inplace=True)
dfs1.dropna(subset=['name'], inplace=True)
dfs1.head()

appid                           name
5    216938  Pieterw test app76 ( 216938 )
6    660010                          test2
7    660130                          test3
23  1872760               Spirit Detective
24  1872770                  赏金游戏 Playtest

In [320]:
gameName='Hitman 3'

In [321]:
#dfs1.loc[dfs1['name']=='test2'].head(3)
to=dfs1.loc[dfs1['name'].str.contains(gameName, case=False)]
to

appid                                               name
1700   1847520                                      HITMAN 3 Demo
1989   1843460        HITMAN 3 Access Pass: HITMAN 1 GOTY Edition
2974   1829590           HITMAN 3 Access Pass: HITMAN 2 Expansion
2975   1829591                             HITMAN 3 - Deluxe Pack
2976   1829592            HITMAN 3 Access Pass: HITMAN 2 Standard
2977   1829593  HITMAN 3 Access Pass: HITMAN 1 Complete First ...
2978   1829594                               HITMAN 3 - VR Access
2979   1829595        HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade
2980   1829596                            HITMAN 3 - Trinity Pack
2981   1829600                    HITMAN 3 - Carpathian Mountains
2982   1829601                                 HITMAN 3 - Mendoza
2983   1829602                               HITMAN 3 - Chongqing
2984   1829603                                  HITMAN 3 - Berlin
2985   1829604                                HITMAN 3 - Dartmoor
2986   1829605                                   HITMAN 3 - Dubai
3035   1829580          HITMAN 3 - Seven Deadly Sins Act 1: Greed
3036   1829581          HITMAN 3 - Seven Deadly Sins Act 2: Pride
3037   1829582          HITMAN 3 - Seven Deadly Sins Act 3: Sloth
3038   1829583           HITMAN 3 - Seven Deadly Sins Act 4: Lust
3039   1829584       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony
3040   1829585           HITMAN 3 - Seven Deadly Sins Act 6: Envy
3041   1829586          HITMAN 3 - Seven Deadly Sins Act 7: Wrath
37455  1659040                                           HITMAN 3

for a in steamApps:
    #if a != None and 'hitman 3' in a['name'].lower():
    if a is not None and 'hitman 3' in a['name'].lower():
        appId = a['appid']
        print(f"Name: {a['name']} - appid: {appId}")

steamAppSingleResponse = requests.get(
    "https://store.steampowered.com/api/appdetails?appids="+str(appId))
steamAppSingle = steamAppSingleResponse.json()

print(steamAppSingle[str(appId)]['data']['price_overview']['final_formatted'])

In [322]:
te=dfs1.loc[dfs1['name'].str.lower().eq(gameName.lower())]
te

appid      name
37455  1659040  HITMAN 3

In [221]:
apa=te.index[0]

In [222]:
#for l in te.head():
te = 'https://store.steampowered.com/api/appdetails?' + 'appids=' + str(apa)
te

'https://store.steampowered.com/api/appdetails?appids=37455'

In [223]:
ta=to
#ta.reset_index()
ta

appid                                               name
1700   1847520                                      HITMAN 3 Demo
1989   1843460        HITMAN 3 Access Pass: HITMAN 1 GOTY Edition
2974   1829590           HITMAN 3 Access Pass: HITMAN 2 Expansion
2975   1829591                             HITMAN 3 - Deluxe Pack
2976   1829592            HITMAN 3 Access Pass: HITMAN 2 Standard
2977   1829593  HITMAN 3 Access Pass: HITMAN 1 Complete First ...
2978   1829594                               HITMAN 3 - VR Access
2979   1829595        HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade
2980   1829596                            HITMAN 3 - Trinity Pack
2981   1829600                    HITMAN 3 - Carpathian Mountains
2982   1829601                                 HITMAN 3 - Mendoza
2983   1829602                               HITMAN 3 - Chongqing
2984   1829603                                  HITMAN 3 - Berlin
2985   1829604                                HITMAN 3 - Dartmoor
2986   1829605                                   HITMAN 3 - Dubai
3035   1829580          HITMAN 3 - Seven Deadly Sins Act 1: Greed
3036   1829581          HITMAN 3 - Seven Deadly Sins Act 2: Pride
3037   1829582          HITMAN 3 - Seven Deadly Sins Act 3: Sloth
3038   1829583           HITMAN 3 - Seven Deadly Sins Act 4: Lust
3039   1829584       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony
3040   1829585           HITMAN 3 - Seven Deadly Sins Act 6: Envy
3041   1829586          HITMAN 3 - Seven Deadly Sins Act 7: Wrath
37455  1659040                                           HITMAN 3

In [230]:
#df['baz'] = df.agg('{0[bar]} is {0[foo]}'.format, axis=1)
#ta['appid']=ta['appid'].astype(str)
ta['url'] = ta.agg('https://store.steampowered.com/api/appdetails?appids={0[appid]}'.format, axis=1)
ta

C:\Users\robbo\AppData\Local\Temp\ipykernel_25804\120048124.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ta['url'] = ta.agg('https://store.steampowered.com/api/appdetails?appids={0[appid]}'.format, axis=1)


appid                                               name  \
1700   1847520                                      HITMAN 3 Demo   
1989   1843460        HITMAN 3 Access Pass: HITMAN 1 GOTY Edition   
2974   1829590           HITMAN 3 Access Pass: HITMAN 2 Expansion   
2975   1829591                             HITMAN 3 - Deluxe Pack   
2976   1829592            HITMAN 3 Access Pass: HITMAN 2 Standard   
2977   1829593  HITMAN 3 Access Pass: HITMAN 1 Complete First ...   
2978   1829594                               HITMAN 3 - VR Access   
2979   1829595        HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade   
2980   1829596                            HITMAN 3 - Trinity Pack   
2981   1829600                    HITMAN 3 - Carpathian Mountains   
2982   1829601                                 HITMAN 3 - Mendoza   
2983   1829602                               HITMAN 3 - Chongqing   
2984   1829603                                  HITMAN 3 - Berlin   
2985   1829604                                HITMAN 3 - Dartmoor   
2986   1829605                                   HITMAN 3 - Dubai   
3035   1829580          HITMAN 3 - Seven Deadly Sins Act 1: Greed   
3036   1829581          HITMAN 3 - Seven Deadly Sins Act 2: Pride   
3037   1829582          HITMAN 3 - Seven Deadly Sins Act 3: Sloth   
3038   1829583           HITMAN 3 - Seven Deadly Sins Act 4: Lust   
3039   1829584       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony   
3040   1829585           HITMAN 3 - Seven Deadly Sins Act 6: Envy   
3041   1829586          HITMAN 3 - Seven Deadly Sins Act 7: Wrath   
37455  1659040                                           HITMAN 3   

                                                     url  
1700   https://store.steampowered.com/api/appdetails?...  
1989   https://store.steampowered.com/api/appdetails?...  
2974   https://store.steampowered.com/api/appdetails?...  
2975   https://store.steampowered.com/api/appdetails?...  
2976   https://store.steampowered.com/api/appdetails?...  
2977   https://store.steampowered.com/api/appdetails?...  
2978   https://store.steampowered.com/api/appdetails?...  
2979   https://store.steampowered.com/api/appdetails?...  
2980   https://store.steampowered.com/api/appdetails?...  
2981   https://store.steampowered.com/api/appdetails?...  
2982   https://store.steampowered.com/api/appdetails?...  
2983   https://store.steampowered.com/api/appdetails?...  
2984   https://store.steampowered.com/api/appdetails?...  
2985   https://store.steampowered.com/api/appdetails?...  
2986   https://store.steampowered.com/api/appdetails?...  
3035   https://store.steampowered.com/api/appdetails?...  
3036   https://store.steampowered.com/api/appdetails?...  
3037   https://store.steampowered.com/api/appdetails?...  
3038   https://store.steampowered.com/api/appdetails?...  
3039   https://store.steampowered.com/api/appdetails?...  
3040   https://store.steampowered.com/api/appdetails?...  
3041   https://store.steampowered.com/api/appdetails?...  
37455  https://store.steampowered.com/api/appdetails?...

In [231]:
ta['url'].values[0]

'https://store.steampowered.com/api/appdetails?appids=1847520'

frames=[]
for index, row in ta.iterrows():
    #print(row["name"], row["url"])
    steamResponse = requests.get(row['url']).json() 
    datac=steamResponse[str(row["appid"])]['data']
    print(datac['name'])
    if "price_overview" in datac:
        print(datac['price_overview']['final_formatted'])
        frames.append(steamSdf)
        s1=ss['1847520']['data']
        steamSdf=pd.json_normalize(s1)
    

single = pd.concat(frames)
steamSdf




In [283]:
karr=[]
keys=['type', 'name', 'steam_appid', 'is_free','controller_support', 
      'dlc', 'detailed_description', 'about_the_game','short_description', 
      'supported_languages', 'developers', 'publishers', 'demos', 'packages', 
      'package_groups', 'categories', 'genres', 'price_overview.currency', 
      'price_overview.initial', 'price_overview.final', 'price_overview.discount_percent', 
      'price_overview.initial_formatted', 'price_overview.final_formatted', 'platforms.windows', 
      'release_date.coming_soon','release_date.date']
for k in keys:
    karr.append([k])

karr
    

[['type'],
 ['name'],
 ['steam_appid'],
 ['is_free'],
 ['controller_support'],
 ['dlc'],
 ['detailed_description'],
 ['about_the_game'],
 ['short_description'],
 ['supported_languages'],
 ['developers'],
 ['publishers'],
 ['demos'],
 ['packages'],
 ['package_groups'],
 ['categories'],
 ['genres'],
 ['price_overview.currency'],
 ['price_overview.initial'],
 ['price_overview.final'],
 ['price_overview.discount_percent'],
 ['price_overview.initial_formatted'],
 ['price_overview.final_formatted'],
 ['platforms.windows'],
 ['release_date.coming_soon'],
 ['release_date.date']]

In [ ]:
muppe=[['type'],
 ['name'],
 ['steam_appid'],
 ['is_free'],
 ['controller_support'],
 ['dlc'],
 ['detailed_description'],
 ['about_the_game'],
 ['short_description'],
 ['supported_languages'],
 ['developers'],
 ['publishers'],
 ['demos'],
 ['packages'],
 ['package_groups'],
 ['categories'],
 ['genres'],
 ['price_overview.currency'],
 ['price_overview.initial'],
 ['price_overview.final'],
 ['price_overview.discount_percent'],
 ['price_overview.initial_formatted'],
 ['price_overview.final_formatted'],
 ['platforms.windows'],
 ['release_date.coming_soon'],
 ['release_date.date']]


In [299]:
frames=[]

In [300]:
for index, row in ta.iterrows():
    #print(row["name"], row["url"])
    steamResponse = requests.get(row['url']).json() 
    datac=steamResponse[str(row["appid"])]['data']
    #if "price_overview" in datac:
    steamSdf=pd.json_normalize(datac)
    steamSdf.keys()
    frames.append(steamSdf)
        
single = pd.concat(frames, ignore_index=True)
single

type                                               name  steam_appid  \
0   demo                                      HITMAN 3 Demo      1847520   
1    dlc        HITMAN 3 Access Pass: HITMAN 1 GOTY Edition      1843460   
2    dlc           HITMAN 3 Access Pass: HITMAN 2 Expansion      1829590   
3    dlc                             HITMAN 3 - Deluxe Pack      1829591   
4    dlc            HITMAN 3 Access Pass: HITMAN 2 Standard      1829592   
5    dlc  HITMAN 3 Access Pass: HITMAN 1 Complete First ...      1829593   
6    dlc                               HITMAN 3 - VR Access      1829594   
7    dlc        HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade      1829595   
8    dlc                            HITMAN 3 - Trinity Pack      1829596   
9    dlc                    HITMAN 3 - Carpathian Mountains      1829600   
10   dlc                                 HITMAN 3 - Mendoza      1829601   
11   dlc                               HITMAN 3 - Chongqing      1829602   
12   dlc                                  HITMAN 3 - Berlin      1829603   
13   dlc                                HITMAN 3 - Dartmoor      1829604   
14   dlc                                   HITMAN 3 - Dubai      1829605   
15   dlc          HITMAN 3 - Seven Deadly Sins Act 1: Greed      1829580   
16   dlc          HITMAN 3 - Seven Deadly Sins Act 2: Pride      1829581   
17   dlc          HITMAN 3 - Seven Deadly Sins Act 3: Sloth      1829582   
18   dlc           HITMAN 3 - Seven Deadly Sins Act 4: Lust      1829583   
19   dlc       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony      1829584   
20   dlc           HITMAN 3 - Seven Deadly Sins Act 6: Envy      1829585   
21   dlc          HITMAN 3 - Seven Deadly Sins Act 7: Wrath      1829586   
22  game                                           HITMAN 3      1659040   

   required_age  is_free controller_support  \
0             0     True               full   
1            18    False               full   
2            18    False               full   
3             0    False               full   
4            18    False               full   
5            18    False               full   
6             0    False               full   
7            18    False               full   
8             0    False               full   
9             0    False               full   
10            0    False               full   
11            0    False               full   
12            0    False               full   
13            0    False               full   
14            0    False               full   
15            0    False               full   
16            0    False               full   
17            0    False               full   
18            0    False               full   
19            0    False               full   
20            0    False               full   
21            0    False               full   
22           18    False               full   

                                 detailed_description  \
0   Become Agent 47 and experience the start of th...   
1   The HITMAN 3 Access Pass system allows you to ...   
2   The HITMAN 3 Access Pass system allows you to ...   
3   The HITMAN 3 - Deluxe Pack adds new content to...   
4   The HITMAN 3 Access Pass system allows you to ...   
5   The HITMAN 3 Access Pass system allows you to ...   
6   Death Awaits. Agent 47 returns in HITMAN 3, th...   
7   The HITMAN 3 Access Pass system allows you to ...   
8   The Trinity Pack includes new items that can b...   
9   The final mission in the World of Assassinatio...   
10  As one of the most well-known wine regions in ...   
11  Under the flickering neon lights of busy cafes...   
12  Head to the outskirts of Berlin, a city known ...   
13  Travel to the historic Thornbridge Manor in Da...   
14  Experience the grandeur and decadence of Dubai...   
15  Act 1: Greed<br />\r\n<br />\r\nHITMAN 3: Seve...   
16  Act 2: Pride<br />\r\n<br />\r\nHITMAN 3: Seve...   
17  Act 3: Sloth<br />\r

In [315]:
steamSE = single.assign(priceSEK=single["price_overview.final"]/100*rate_sek)
steamSE

type                                               name  steam_appid  \
0   demo                                      HITMAN 3 Demo      1847520   
1    dlc        HITMAN 3 Access Pass: HITMAN 1 GOTY Edition      1843460   
2    dlc           HITMAN 3 Access Pass: HITMAN 2 Expansion      1829590   
3    dlc                             HITMAN 3 - Deluxe Pack      1829591   
4    dlc            HITMAN 3 Access Pass: HITMAN 2 Standard      1829592   
5    dlc  HITMAN 3 Access Pass: HITMAN 1 Complete First ...      1829593   
6    dlc                               HITMAN 3 - VR Access      1829594   
7    dlc        HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade      1829595   
8    dlc                            HITMAN 3 - Trinity Pack      1829596   
9    dlc                    HITMAN 3 - Carpathian Mountains      1829600   
10   dlc                                 HITMAN 3 - Mendoza      1829601   
11   dlc                               HITMAN 3 - Chongqing      1829602   
12   dlc                                  HITMAN 3 - Berlin      1829603   
13   dlc                                HITMAN 3 - Dartmoor      1829604   
14   dlc                                   HITMAN 3 - Dubai      1829605   
15   dlc          HITMAN 3 - Seven Deadly Sins Act 1: Greed      1829580   
16   dlc          HITMAN 3 - Seven Deadly Sins Act 2: Pride      1829581   
17   dlc          HITMAN 3 - Seven Deadly Sins Act 3: Sloth      1829582   
18   dlc           HITMAN 3 - Seven Deadly Sins Act 4: Lust      1829583   
19   dlc       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony      1829584   
20   dlc           HITMAN 3 - Seven Deadly Sins Act 6: Envy      1829585   
21   dlc          HITMAN 3 - Seven Deadly Sins Act 7: Wrath      1829586   
22  game                                           HITMAN 3      1659040   

   required_age  is_free controller_support  \
0             0     True               full   
1            18    False               full   
2            18    False               full   
3             0    False               full   
4            18    False               full   
5            18    False               full   
6             0    False               full   
7            18    False               full   
8             0    False               full   
9             0    False               full   
10            0    False               full   
11            0    False               full   
12            0    False               full   
13            0    False               full   
14            0    False               full   
15            0    False               full   
16            0    False               full   
17            0    False               full   
18            0    False               full   
19            0    False               full   
20            0    False               full   
21            0    False               full   
22           18    False               full   

                                 detailed_description  \
0   Become Agent 47 and experience the start of th...   
1   The HITMAN 3 Access Pass system allows you to ...   
2   The HITMAN 3 Access Pass system allows you to ...   
3   The HITMAN 3 - Deluxe Pack adds new content to...   
4   The HITMAN 3 Access Pass system allows you to ...   
5   The HITMAN 3 Access Pass system allows you to ...   
6   Death Awaits. Agent 47 returns in HITMAN 3, th...   
7   The HITMAN 3 Access Pass system allows you to ...   
8   The Trinity Pack includes new items that can b...   
9   The final mission in the World of Assassinatio...   
10  As one of the most well-known wine regions in ...   
11  Under the flickering neon lights of busy cafes...   
12  Head to the outskirts of Berlin, a city known ...   
13  Travel to the historic Thornbridge Manor in Da...   
14  Experience the grandeur and decadence of Dubai...   
15  Act 1: Greed<br />\r\n<br />\r\nHITMAN 3: Seve...   
16  Act 2: Pride<br />\r\n<br />\r\nHITMAN 3: Seve...   
17  Act 3: Sloth<br />\r

In [318]:
steamSE.filter(['name','type','controller_support',
       'price_overview.final',
       'price_overview.discount_percent',
       'price_overview.final_formatted','priceSEK'],axis=1)

name  type  \
0                                       HITMAN 3 Demo  demo   
1         HITMAN 3 Access Pass: HITMAN 1 GOTY Edition   dlc   
2            HITMAN 3 Access Pass: HITMAN 2 Expansion   dlc   
3                              HITMAN 3 - Deluxe Pack   dlc   
4             HITMAN 3 Access Pass: HITMAN 2 Standard   dlc   
5   HITMAN 3 Access Pass: HITMAN 1 Complete First ...   dlc   
6                                HITMAN 3 - VR Access   dlc   
7         HITMAN 3 Access Pass: HITMAN 1 GOTY Upgrade   dlc   
8                             HITMAN 3 - Trinity Pack   dlc   
9                     HITMAN 3 - Carpathian Mountains   dlc   
10                                 HITMAN 3 - Mendoza   dlc   
11                               HITMAN 3 - Chongqing   dlc   
12                                  HITMAN 3 - Berlin   dlc   
13                                HITMAN 3 - Dartmoor   dlc   
14                                   HITMAN 3 - Dubai   dlc   
15          HITMAN 3 - Seven Deadly Sins Act 1: Greed   dlc   
16          HITMAN 3 - Seven Deadly Sins Act 2: Pride   dlc   
17          HITMAN 3 - Seven Deadly Sins Act 3: Sloth   dlc   
18           HITMAN 3 - Seven Deadly Sins Act 4: Lust   dlc   
19       HITMAN 3 - Seven Deadly Sins Act 5: Gluttony   dlc   
20           HITMAN 3 - Seven Deadly Sins Act 6: Envy   dlc   
21          HITMAN 3 - Seven Deadly Sins Act 7: Wrath   dlc   
22                                           HITMAN 3  game   

   controller_support  price_overview.final  price_overview.discount_percent  \
0                full                   NaN                              NaN   
1                full                2999.0                              0.0   
2                full                3499.0                              0.0   
3                full                3499.0                              0.0   
4                full                3999.0                              0.0   
5                full                9999.0                              0.0   
6                full                9999.0                              0.0   
7                full                 999.0                              0.0   
8                full                   NaN                              NaN   
9                full                9999.0                              0.0   
10               full                9999.0                              0.0   
11               full                9999.0                              0.0   
12               full                9999.0                              0.0   
13               full                9999.0                              0.0   
14               full                9999.0                              0.0   
15               full                 499.0                              0.0   
16               full                 499.0                              0.0   
17               full                 499.0                              0.0   
18               full                 499.0                              0.0   
19               full                 499.0                              0.0   
20               full                 499.0                              0.0   
21               full                 499.0                              0.0   
22               full                5999.0                              0.0   

   price_overview.final_formatted     priceSEK  
0                             NaN          NaN  
1                          29,99€   328.396498  
2                          34,99€   383.147498  
3                          34,99€   383.147498  
4                          39,99€   437.898498  
5                          99,99€  1094.910498  
6                          99,99€  1094.910498  
7                           9,99€   109.392498  
8                             NaN          NaN  
9                          99,99€  1094.910498  
10                         99,99€  1094.910498  
11                         

In [267]:
surl='https://store.steampowered.com/api/appdetails?appids=1659040'
ss = requests.get(surl).json()
s1=ss['1659040']['data']
steamSdf=pd.json_normalize(s1)
steamSdf.keys()

Index(['type', 'name', 'steam_appid', 'required_age', 'is_free',
       'controller_support', 'dlc', 'detailed_description', 'about_the_game',
       'short_description', 'supported_languages', 'header_image', 'website',
       'legal_notice', 'developers', 'publishers', 'demos', 'packages',
       'package_groups', 'categories', 'genres', 'screenshots', 'movies',
       'background', 'background_raw', 'pc_requirements.minimum',
       'pc_requirements.recommended', 'mac_requirements.minimum',
       'mac_requirements.recommended', 'linux_requirements.minimum',
       'linux_requirements.recommended', 'price_overview.currency',
       'price_overview.initial', 'price_overview.final',
       'price_overview.discount_percent', 'price_overview.initial_formatted',
       'price_overview.final_formatted', 'platforms.windows', 'platforms.mac',
       'platforms.linux', 'metacritic.score', 'metacritic.url',
       'recommendations.total', 'achievements.total',
       'achievements.highlighted